services/discordService.js

```diff
@@ -0,0 +1,30 @@
+const jwt = require("jsonwebtoken");
+
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+
+const getDiscordMembers = async () => {
+  const authToken = jwt.sign({}, config.get("rdsServerlessBot.rdsServerLessPrivateKey"), {
+    algorithm: "RS256",
+    expiresIn: config.get("rdsServerlessBot.ttl"),
+  });
+
+  try {
+    const response = await fetch(`${DISCORD_BASE_URL}/discord-members`, {
+      method: "GET",
+      headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+    })
+      .then((response, err) => {
+        return response.json();
+      })
+      .then((data) => {
+        return data;
+      });
```

Comment:
can you please confirm if this `.then` is needed. as it is just returning the data which it is receiving. Removing this should still make the code work.

Reply:
yes i tried it normally but id did not work as our cfw sends nested response, which wasn't accessible normally.

Also, could you provide reference where it sayswe should not use `.then`?

Reply:
Certainly! While there are no strict rules about using `async` and `await` in place of promise chaining,  using async await makes code easier to understand. If you see this mdn doc [here](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises) it says 
```note
The async and await keywords make it easier to build an operation from a series of consecutive asynchronous function calls, avoiding the need to create explicit promise chains, and allowing you to write code that looks just like synchronous code.
```